libraries/chain/controller.cpp

```diff
@@ -3999,15 +3986,95 @@ struct controller_impl {
       bsp->verify_qc(qc_proof);
    }
 
+   // Verify QC block extension and finality block header extension on a Legacy
+   // or Transition block.
+   void verify_legacy_block_exts( const signed_block_ptr& b, const block_header_state_legacy& prev ) {
+      uint32_t block_num = b->block_num();
+
+      auto block_exts = b->validate_and_extract_extensions();
+      auto qc_ext_id = quorum_certificate_extension::extension_id();
+      bool qc_extension_present = block_exts.count(qc_ext_id) != 0;
+      EOS_ASSERT( !qc_extension_present,
+                  block_validate_exception,
+                  "Legacy or Transition block #${b} includes a QC block extension",
```

Comment:
This is not called for `Transition blocks`.

Reply:
The function is called before a block becomes proposer Savanna block. It can be a transition block.

Reply:
Yes, sorry. My mistake.